iosApp/iosApp/Pages/Map/HomeMapViewHandlerExtension.swift

```diff
@@ -101,7 +101,9 @@ extension HomeMapView {
         leaveVehiclesChannel()
         vehiclesRepository.connect(routeId: routeId, directionId: directionId) { outcome in
             if case let .ok(result) = onEnum(of: outcome) {
-                vehiclesData = Array(result.data.vehicles.values)
+                if let departures = nearbyVM.departures {
+                    vehiclesData = Array(departures.filterVehiclesByUpcoming(vehicles: result.data).values)
+                }
```

Comment:
suggestion(if-minor): This looks good but seems pretty hard to test as-is. That may be a case for moving this logic into the VM - imagining a func like `handleNewVehicles(departures, vehicles)` that could be tested independently and confirms that `vehiclesData` is set to the filtered list.

Reply:
I'm gonna hold off on doing this right now 🙏  It might make sense to roll into the changes when we split up the nearby VM?